### ISSUE TITLE
[BUG] `assess` with `vmap_combinator` is broken

### ISSUE THREAD

**Issue by ahiser1117** [genjax-community/genjax#1114]
**Describe the bug**

When creating a `GenerativeFunction` with the `vmap_combinator` I am unable to `assess` a given `ChoiceMap`.

**To Reproduce**

Using the `vmap_combinator` model shown in the docs:
```python
@genjax.vmap_combinator(in_axes=(0,))
@genjax.gen
def mapped(x):
    noise1 = genjax.normal(0.0, 1.0) @ "noise1"
    noise2 = genjax.normal(0.0, 1.0) @ "noise2"
    return x + noise1 + noise2
```
Generate a sample by simulating and extracting the sample from the `Trace`. Then attempt to `assess`
```python
key = jax.random.PRNGKey(314159)
arr = jnp.ones(100)
tr = jax.jit(mapped.simulate)(key, (arr,))

mapped.assess(tr.get_sample(), (arr,))
```
And receive the following error on the last line.
```
ValueError: vmap in_axes must be an int, None, or a tuple of entries corresponding to the positional arguments passed to the function, but got len(in_axes)=4, len(args)=2
```
Traceback indicates this is an issue from line 291 in `genjax/_src/generative_functions/combinators/vmap_combinator.py`

**Expected behavior**
The `ChoiceMap` is properly scored using the `GenerativeFunction`

**Execution environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Python version: 3.12.3
 - GenJAX version: 0.4.0 (latest `main` commit as of 10:45 5/22) (16753f7398c99f097a25a381403405832fd8e5ca)

**Comment by ahiser1117** [comment]
I was able to get it to work for my use case by changing lines 291-293 in `genjax/src/genjax/_src/generative_functions/combinators/vmap_combinator.py` to

```python
scores, retvals = jax.vmap(_assess, in_axes=(0, self.in_axes))(
    idx_array, args
)
```

**Comment by femtomc** [comment]
Will be closed by #1115.

**Comment by sritchie** [comment]
Note that this will sync back
